drivers/counter/counter_mcux_pit.c

```diff
@@ -0,0 +1,259 @@
+	PIT_ClearStatusFlags(config->base, config->pit_channel, flags);
+	if (data->top_callback) {
+		data->top_callback(dev, data->top_user_data);
+	}
+	if (data->alarm_callback) {
+		PIT_StopTimer(config->base, config->pit_channel);
+		mcux_pit_get_value(dev, &current);
+		data->alarm_callback(dev, config->pit_channel, current,
+				     data->alarm_user_data);
+	}
+}
+
+static int mcux_pit_set_alarm(struct device *dev, uint8_t chan_id,
+			      const struct counter_alarm_cfg *alarm_cfg)
+{
+	const struct mcux_pit_config *config = dev->config_info;
+	struct mcux_pit_data *data = dev->driver_data;
+
+	uint32_t ticks = alarm_cfg->ticks;
+
+	if (chan_id != DT_PROP(DT_DRV_INST(0), pit_channel)) {
+		LOG_ERR("Invalid channel id");
+		return -EINVAL;
+	}
+
+	if (ticks > mcux_pit_get_top_value(dev)) {
+		LOG_ERR("Invalid tciks");
+		return -EINVAL;
+	}
+
+	PIT_StopTimer(config->base, chan_id);
+	PIT_SetTimerPeriod(config->base, chan_id, ticks);
+	data->alarm_callback = alarm_cfg->callback;
+	data->alarm_user_data = alarm_cfg->user_data;
+	LOG_DBG("set alarm to %d", ticks);
+	PIT_StartTimer(config->base, chan_id);
+	return 0;
+}
+
+static int mcux_pit_cancel_alarm(struct device *dev, uint8_t chan_id)
+{
+	const struct mcux_pit_config *config = dev->config_info;
+	struct mcux_pit_data *data = dev->driver_data;
+
+	if (chan_id != DT_PROP(DT_DRV_INST(0), pit_channel)) {
+		LOG_ERR("Invalid channel id");
+		return -EINVAL;
+	}
+
+	PIT_DisableInterrupts(config->base, chan_id, kPIT_TimerInterruptEnable);
+	PIT_StopTimer(config->base, chan_id);
+	data->alarm_callback = NULL;
+
+	return 0;
+}
+
+static int mcux_pit_init(struct device *dev)
+{
+	struct mcux_pit_config *config =
+		(struct mcux_pit_config *)dev->config_info;
+	pit_config_t pit_config;
+
+	PIT_GetDefaultConfig(&pit_config);
+	pit_config.enableRunInDebug = config->enableRunInDebug;
+
+	PIT_Init(config->base, &pit_config);
+
+	config->irq_config_func(dev);
+
+	PIT_SetTimerPeriod(config->base, config->pit_channel,
+			   USEC_TO_COUNT(DT_PROP(DT_DRV_INST(0), pit_period),
+					 CLOCK_GetFreq(kCLOCK_BusClk)));
+	config->info.freq = CLOCK_GetFreq(kCLOCK_BusClk);
+
+	return 0;
+}
+
+static const struct counter_driver_api mcux_pit_driver_api = {
+	.start = mcux_pit_start,
+	.stop = mcux_pit_stop,
+	.get_value = mcux_pit_get_value,
+	.set_top_value = mcux_pit_set_top_value,
+	.set_alarm = mcux_pit_set_alarm,
+	.cancel_alarm = mcux_pit_cancel_alarm,
+	.get_pending_int = mcux_pit_get_pending_int,
+	.get_top_value = mcux_pit_get_top_value,
+	.get_max_relative_alarm = mcux_pit_get_max_relative_alarm,
+};
+
+/*
+ * This driver is single-instance. If the devicetree contains multiple
+ * instances, this will fail and the driver needs to be revisited.
+ */
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) <= 1,
+	     "unsupported pit instance");
+
+static struct mcux_pit_data mcux_pit_data_0;
+
+static void mcux_pit_irq_config_0(struct device *dev);
+
+static struct mcux_pit_config mcux_pit_config_0 = {
```

Comment:
add `const`

Reply:
ok

Reply:
That means we have to hard code it in dts by adding a `clock-frequency` property to `dts/bindings/rtc/nxp,kinetis-pit.yaml`. The microchip rtc driver does this.

cc: @nordic-krch @pabigot

Reply:
ok